src/renderer/ThemeBackend.js

```diff
@@ -0,0 +1,225 @@
+var Color = require('color')
+
+function changeContrast (colorString, factor) {
+  // TODO make the black check code work
+  const color = Color(colorString).hex() === '#000000' ? Color('#010101') : Color(colorString)
+  if (color.isDark()) {
+    console.log('dark')
+    return color.lighten(factor).rgb().string()
+  } else if (color.isLight()) {
+    console.log('light')
+    return color.darken(factor).rgb().string()
+  }
+}
+
+function invertColor (colorString) {
+  const color = Color(colorString)
+  return color.negate().rgb().string()
+}
+
+function changeSaturation (colorString, factor) {
+  const color = Color(colorString)
+  if (factor < 0) {
+    return color.desaturate(factor *= -1).rgb().string()
+  } else if (factor > 0) {
+    return color.saturate(factor).rgb().string()
+  } else {
+    return color.rgb().string()
+  }
+}
+
+function undefinedGuard (value, func) {
+  if (typeof value === 'undefined') return undefined
+  else return func(value)
+}
+
+export function ThemeDataBuilder (theme) {
+  // todo resolve all strings to be dependent on vars of the highLevelObject
+  // Its ok when highLevelObject is missing some properties
+  // and the returned object misses some values as result,
+  // because it gets merged with the default theme later anyway
+  /// Don't use color directly here
+  console.log(theme.textNavBar, changeContrast(theme.textNavBar, 0.3))
```

Comment:
log.debug

Reply:
same here this is a debug statement that is deprecated now